cmd/boostd/index.go

```diff
@@ -66,13 +62,46 @@ var indexProvListMultihashesCmd = &cli.Command{
 		}
 		defer closer()
 
-		// get list of multihashes
-		mhs, err := napi.BoostIndexerListMultihashes(ctx, propCid)
+		if cctx.Args().Len() != 1 {
+			return fmt.Errorf("must specify only one proposal CID / deal UUID")
+		}
+
+		id := cctx.Args().Get(0)
+
+		var proposalCid cid.Cid
+		var mhs []multihash.Multihash
+		dealUuid, err := uuid.Parse(id)
+		if err != nil {
+			propCid, err := cid.Decode(id)
+			if err != nil {
+				return fmt.Errorf("could not parse '%s' as deal uuid or proposal cid", id)
+			}
+			proposalCid = propCid
+		}
+
+		if !proposalCid.Defined() {
+			contextID, err := dealUuid.MarshalBinary()
+			if err != nil {
+				return fmt.Errorf("parsing UUID to bytes: %w", err)
+			}
+			mhs, err = napi.BoostIndexerListMultihashes(ctx, contextID)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Found %d multihashes for deal with ID %s:\n", len(mhs), id)
+			for _, mh := range mhs {
+				fmt.Println("  " + mh.String())
+			}
+
+			return nil
+		}
+
+		mhs, err = napi.BoostIndexerListMultihashes(ctx, proposalCid.Bytes())
 		if err != nil {
 			return err
 		}
 
-		fmt.Printf("Found %d multihashes for deal with proposal cid %s:\n", len(mhs), propCid)
+		fmt.Printf("Found %d multihashes for deal with ID %s:\n", len(mhs), id)
```

Comment:
any reason this couldn't be JSON? It would be more appreciated for system tools. 
```json
{
"multihashes":  {
      "dealIdentifier1": [ "firstHash", "secondhash"],
      "dealIdentifier2": [ "thirdHash", "fourthHash"],
      "dealIdentifier3": [ ],
      }
}
```

it's as easy as 
```go
e := json.NewEncoder(os.Stdout)
e.SetIndent("  ",  "  ")
e.Encode(map[string]any{ "multihashes": map_Built_Iteratively})
```

Reply:
We have a dedicated method in GraphQL API for this. The output there is a JSON structure with array. containing multihash and corresponding CID as well.